File: Exercise/LuoGu/Code_2023_7_21/P1152.java
package Code_2023_7_21;

import java.util.Scanner;

public class P1152 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int[] arr = new int[n];
        boolean[] arrCheck = new boolean[n];

        int diff;
        arr[0] = sc.nextInt();

        for (int i = 1; i < n; i++) {
            arr[i] = sc.nextInt();
            diff = arr[i] - arr[i - 1];
            diff = diff > 0 ? diff : -diff;
            if (diff < n) {
                arrCheck[diff] = true;
            }
        }

        boolean isJolly = true;
        for (int i = 1; i < n; i++) {
            if (!arrCheck[i]) {
                isJolly = false;
                break;
            }
        }

        if (isJolly) {
            System.out.println("Jolly");
        } else {
            System.out.println("Not jolly");
        }

        sc.close();

    }
}

File: Exercise/LuoGu/Code_2023_7_21/P1152.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>

using namespace std;

int main(void)
{

    int n;
    cin >> n;
    int arr[n];
    int diff;
    cin >> arr[0];

    int arr_check[n];
    for (int i = 0; i < n; i++)
    {
        arr_check[i] = -1;
    }

    for (int i = 1; i < n; i++)
    {
        cin >> arr[i];
        diff = arr[i] - arr[i - 1];
        diff = diff > 0 ? diff : -diff;
        if (diff < n)
        {
            arr_check[diff] = 1;
        }
    }

    bool is_jolly = true;

    for (int i = 1; i < n; i++)
    {
        if (arr_check[i] == -1)
        {
            cout << "Not jolly" << endl;
            is_jolly = false;
            break;
        }
    }
    if (is_jolly)
    {
        cout << "Jolly" << endl;
    }

    return 0;
}